### PR TITLE
nix: add static nix via `pkgs.nixStatic`

### PR DESCRIPTION
**Pull request by DavHau**
###### Motivation for this change
Having a static nix build will open up more possibilities to create alternative and simpler installation mechanisms for nix.

Also I believe this is very useful for other restricted scenarios where a proper installation is just not feasible.

My personal motivation behind this is to create a portable nix distribution to distribute via pypi and conda.

###### Things done
added attribute `pkgs.nixStatic`

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kisik21** [comment]
Is there a reason why can't you use `pkgsStatic.nix` and need an alias for it?

**Comment by xaverdh** [comment]
> Is there a reason why you can't use `pkgsStatic.nix` and need an alias for it?

so its built by hydra probably..

**Comment by DavHau** [comment]
I'd like to have it built by hydra due to caching and also to increase the reliability. If it's not built by hydra, nobody will notice if it breaks.

**Comment by DavHau** [comment]
Can we merge this?

**Review by r-burns — APPROVED** [review]
I'm +1 on this, I think it would be nice to have

**Comment by DavHau** [comment]
I just found out that the flake for nix 2.4 offers a 'nix-static' package which is already built by hydra. For my use case, this is sufficient.

We could still merge this PR to have a stable nix release built as well.

**Review by tomberek — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
@DavHau The build sadly fails because the default curl now included brotli support which does not static compile.

If you can fix that I am for a merge.

**Comment by DavHau** [comment]
@SuperSandro2000 the build succeeds now.

File: pkgs/top-level/static.nix [hunk 1]
```diff
@@ -93,27 +93,36 @@ in {
...
-    # Don’t use new stdenv zlib because
-    # it doesn’t like the --disable-shared flag
+  boost = super.boost.override {
+    # Don’t use new stdenv for boost because it doesn’t like the
+    # --disable-shared flag
     stdenv = super.stdenv;
   };
+
+  curl = super.curl.override {
+    brotliSupport = false;
```
Comment (SuperSandro2000) [inline pkgs/top-level/static.nix:103]: Maybe we should comment this.

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/top-level/static.nix (hunk 1 above)
Comment (DavHau) [reply pkgs/top-level/static.nix:104]: @SuperSandro2000, I added a comment about the disabled brotli. Ready to merge?

**Pull request merged**